ci/scripts/jni_build.sh

```diff
@@ -50,11 +50,15 @@ Darwin)
 esac
 
 : "${ARROW_JAVA_BUILD_TESTS:=${ARROW_BUILD_TESTS:-ON}}"
+: "${ARROW_VCPKG:=OFF}"
 : "${CMAKE_BUILD_TYPE:=release}"
 read -ra EXTRA_CMAKE_OPTIONS <<<"${JAVA_JNI_CMAKE_ARGS:-}"
+# Must set ARROW_VCPKG because upstream Findutf8proc.cmake checks that to
+# determine what to do
 cmake \
   -S "${source_dir}" \
   -B "${build_dir}" \
+  -DARROW_VCPKG="${ARROW_VCPKG}" \
```

Comment:
Can we use `ARROW_DEPENDENCY_SOURCE=VCPKG` instead? It sets `ARROW_VCPKG` internally.

Reply:
I tried this and it doesn't seem to work: I don't think the Java JNI CMake code has anything to detect this. Also, `Findutf8proc.cmake` installed by Arrow's CMake checks `ARROW_VCPKG`, but I don't see any other code installed to set this.

Reply:
Ah, OK. I'll fix it in apache/arrow.

Reply:
https://github.com/apache/arrow/pull/46565 will fix this.